src/scenario.py

```diff
@@ -167,255 +166,284 @@ def run(self, strategy_name, options):
+                            bat_dict = {batName: max(values) for batName, values in
+                                        batteryLevels[gcID].items()}
+                            bat_dict.update({
+                                "unit": "kWh",
+                                "info": "Maximum stored energy in each battery by name"
+                            })
+                            json_results["max. stored energy in batteries"] = bat_dict
+
+                    # charging cycles
+                    # stationary batteries
+                    total_bat_cap = 0
+                    for batID, battery in self.constants.batteries.items():
+                        if self.constants.batteries[batID].parent == gcID:
+                            if battery.capacity > 2**63:
+                                # unlimited capacity
+                                max_cap = max(batteryLevels[gcID][batID])
+                                print("Battery {} is unlimited, set capacity to {} kWh".format(
+                                    batID, max_cap))
+                                total_bat_cap += max_cap
+                            else:
+                                total_bat_cap += battery.capacity
+                    if total_bat_cap:
+                        total_bat_energy = 0
+                        for loads in extLoads[gcID]:
+                            for batID in self.constants.batteries.keys():
+                                if self.constants.batteries[batID].parent == gcID:
+                                    total_bat_energy += max(loads.get(batID, 0), 0) / stepsPerHour
+                        json_results["stationary battery cycles"] = {
+                            "value": total_bat_energy / total_bat_cap,
+                            "unit": None,
+                            "info": "Number of load cycles of stationary batteries (averaged)"
+                        }
+                    # vehicles
+                    total_car_cap[gcID] = sum([v.battery.capacity for v in
+                                               self.constants.vehicles.values()])
+                    total_car_energy[gcID] = sum([sum(map(
+                        lambda v: max(v, 0), r["commands"].values())) for r in results])
+                    json_results["all vehicle battery cycles"] = {
+                        "value": total_car_energy[gcID]/total_car_cap[gcID],
+                        "unit": None,
+                        "info": "Number of load cycles per vehicle (averaged)"
+                    }
+
+                    if options.get("save_results", False):
+                        # write to file
+                        file_name = options['save_results'].split(".")[0] + f"_{gcID}." + options[
+                            'save_results'].split(".")[1]
+                        with open(file_name, 'w') as results_file:
+                            json.dump(json_results, results_file, indent=2)
```

Comment:
assuming there is only one gc, how does the output differ from before? Can the old evaluation tools still be used?

Reply:
the name of the file is changed to.g. "simulation_{gc_name}.json" and "simulation_{gc_name}.json"